commit 56d2cf6f1ef008ab6cb50545af9b1d8a5c1ece81
Author: Ondřej Surý <ondrej@isc.org>
Date:   Mon Aug 24 10:15:07 2020 +0200

    Print diagnostics on dns_name_issubdomain() failure in fctx_create()
    
    Log diagnostic message when dns_name_issubdomain() in the fctx_create()
    when the resolver is qname minimizing and forwarding at the same time.
    
    (cherry picked from commit 0a22024c270a38a54f0d51621a046b726df158c0)

diff --git a/lib/dns/resolver.c b/lib/dns/resolver.c
index 8a36693a9a..b37e756c53 100644
--- a/lib/dns/resolver.c
+++ b/lib/dns/resolver.c
@@ -5122,7 +5122,14 @@ fctx_create(dns_resolver_t *res, const dns_name_t *name, dns_rdatatype_t type,
 
 	log_ns_ttl(fctx, "fctx_create");
 
-	INSIST(dns_name_issubdomain(&fctx->name, &fctx->domain));
+	if (!dns_name_issubdomain(&fctx->name, &fctx->domain)) {
+		dns_name_format(&fctx->domain, buf, sizeof(buf));
+		UNEXPECTED_ERROR(__FILE__, __LINE__,
+				 "'%s' is not subdomain of '%s'", fctx->info,
+				 buf);
+		result = ISC_R_UNEXPECTED;
+		goto cleanup_fcount;
+	}
 
 	fctx->qmessage = NULL;
 	result = dns_message_create(mctx, DNS_MESSAGE_INTENTRENDER,
